### ISSUE TITLE
quarterly report

### ISSUE THREAD

**Issue by bin123apple** [HPC-Fortran2CPP/Fortran2Cpp#49]
Also, I’m interested in detail such as:
1. Where do you get your tuning data from?
2. How much data points do you have? How large and complex are those examples?
3. At what rate did you see improvements when tuning an existing model?
4. How did you verify correctness?
5. Did you measure performance, and if so, what were the results?

**Comment by bin123apple** [comment]
@chunhualiao @peihunglin @LChenGit 

I just wrote some potential answers for these five questions. If there is any questions, please let me know. 

**For 1:**

Our original Fortran code comes from the Fortran code in `codeparrot/github-code`: https://huggingface.co/datasets/codeparrot/github-code

**For 2:**

We have a total of $2529$ data points (`data/F2C_dialogue_25K.json`), each corresponding to a multi-turn dialogue. The number of dialogue turns ranges from $5$ to $17$. These dialogues document the process of translating and refining the original Fortran code. The final Fortran-C++ code pairs are located in `data/readable_format`.

**For 3:**

As the training steps increase, the changes in training loss are shown in the `training_log` folder.

**For 4:**

To ensure the accuracy of the dataset, we performed checks on the code within the dataset using Compilation Error Checking, Execution Error Checking, and Paired Output Checking.

***Compilation Error Checking***

The Fortran and C++ code, with unit test cases inserted, are checked using the g++ and gfortran compilers.

***Execution Error Checking***

The compilable code is executed to check for any errors at this step.

***Paired Output Checking***

In this step, we ask GPT-4 to check if the outputs of the Fortran and C++ are the same. 

**For 5:**

We used four metrics to evaluate the model's performance on the Fortran-C++ translation task, including CodeBLEU Score, Compilation Check, Execution Test Evaluation, and Manual Investigation. The results are shown in the overleaf table 1 and table 2.

**Comment by bin123apple** [comment]
> @chunhualiao @peihunglin @LChenGit
> 
> I just wrote some potential answers for these five questions. If there is any questions, please let me know.
> 
> **For 1:**
> 
> Our original Fortran code comes from the Fortran code in `codeparrot/github-code`: https://huggingface.co/datasets/codeparrot/github-code
> 
> **For 2:**
> 
> We have a total of 2529 data points (`data/F2C_dialogue_25K.json`), each corresponding to a multi-turn dialogue. The number of dialogue turns ranges from 5 to 17 . These dialogues document the process of translating and refining the original Fortran code. The final Fortran-C++ code pairs are located in `data/readable_format`.
> 
> **For 3:**
> 
> As the training steps increase, the changes in training loss are shown in the `training_log` folder.
> 
> **For 4:**
> 
> To ensure the accuracy of the dataset, we performed checks on the code within the dataset using Compilation Error Checking, Execution Error Checking, and Paired Output Checking.
> 
> _**Compilation Error Checking**_
> 
> The Fortran and C++ code, with unit test cases inserted, are checked using the g++ and gfortran compilers.
> 
> _**Execution Error Checking**_
> 
> The compilable code is executed to check for any errors at this step.
> 
> _**Paired Output Checking**_
> 
> In this step, we ask GPT-4 to check if the outputs of the Fortran and C++ are the same.
> 
> **For 5:**
> 
> We used four metrics to evaluate the model's performance on the Fortran-C++ translation task, including CodeBLEU Score, Compilation Check, Execution Test Evaluation, and Manual Investigation. The results are shown in the overleaf table 1 and table 2.

Hey Team @chunhualiao @peihunglin @LChenGit !  Please let me know if you need other new information or you have questions related to this reply for the presentation on Tuesday.

**Comment by chunhualiao** [comment]
@bin123apple  Your answer to Q3 is not clear: At what rate did you see improvements when tuning an existing model?

" the changes in training loss are shown in the training_log folder."  Your answer raises more questions than addressing a question.  How many files are in the folder and how many lines each file has? which file and which line should I look at?  What changes are you talking about? How should I extract a rate from the changes of this folder?  

The expected answer may be a ratio between the fine-tuned vs. baseline model using some accuracy metrics, such as your Table 1 or Table 2 metrics.  You can compute the ratio in the tables, like what we did in last year's paper.

**Comment by bin123apple** [comment]
OKOK, I thought the "**rate**" means "**speed**".

Here is the 

> ratio between the fine-tuned vs. baseline model

For F2C++ Dataset:

| Model                | CodeBLEU Ratio (w ft/ wo ft) | Compilation Ratio  (w ft/ wo ft)| 
|:----------------------:|:----------------:|:-------------------:|
| DeepSeek-Coder 6.7B   | $1.5521$         | ∞               | 
| Codellama 13B        | $1.7933$         | ∞               |
| StarCoder 15.5B      | $1.7283$         | ∞               | 

For translated HumanEval: 

| Model                | CodeBLEU Ratio (w ft/ wo ft) | Compilation Ratio (w ft/ wo ft) | Execution Ratio (w ft/ wo ft) |
|:----------------------:|:----------------:|:-------------------:|:-----------------:|
| DeepSeek-Coder 6.7B   | $3.1104$         | ∞               | ∞             |
| Codellama 13B        | $0.8013$         | ∞               | ∞             |
| StarCoder 15.5B      | $3.3087$         | ∞               | ∞             |

**Comment by chunhualiao** [comment]
$ means us dollar by default. Why do you use $ to include numbers?

Please add the non-infinite ratios into the paper' tables.

**Comment by bin123apple** [comment]
Finish quarterly report.
